test/e2e/setup.go

```diff
@@ -75,6 +85,149 @@ func skipIfNotInDevelopmentEnv() {
 	}
 }
 
+func TakeScreenshot(wd WebDriver, e error) {
+	log.Infof("Error : %s", e.Error())
+	log.Info("Taking Screenshot and saving page source")
+	imageBytes, err := wd.Screenshot()
+	if err != nil {
+		panic(err)
+	}
+
+	imageData, err := png.Decode(bytes.NewReader(imageBytes))
+	if err != nil {
+		panic(err)
+	}
+
+	sourceString, err := wd.PageSource()
+	if err != nil {
+		panic(err)
+	}
+
+	errorString := strings.ReplaceAll(e.Error(), " ", "_")
+
+	imagePath := "./" + errorString + ".png"
+	sourcePath := "./" + errorString + ".html"
+
+	imageAbsPath, err := filepath.Abs(imagePath)
+	if err != nil {
+		panic(err)
+	}
+	sourceAbsPath, err := filepath.Abs(sourcePath)
+	if err != nil {
+		panic(err)
+	}
+
+	image, err := os.Create(imageAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	source, err := os.Create(sourceAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	err = png.Encode(image, imageData)
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = source.WriteString(sourceString)
+	if err != nil {
+		panic(err)
+	}
+
+	err = image.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	err = source.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	log.Infof("Screenshot saved to %s", imageAbsPath)
+	log.Infof("Page Source saved to %s", sourceAbsPath)
+
+	panic(e)
+}
+
+func adminPortalSessionSetup() (string, *WebDriver) {
+	const (
+		port = 4444
+	)
+
+	os.Setenv("SE_SESSION_REQUEST_TIMEOUT", "9000")
+
+	caps := Capabilities{
+		"browserName":         "MicrosoftEdge",
+		"acceptInsecureCerts": true,
+	}
+	wd := WebDriver(nil)
+
+	_, err := url.ParseRequestURI("https://localhost:4444")
+	if err != nil {
+		panic(err)
+	}
+
+	for i := 0; i < 10; i++ {
+		wd, err = NewRemote(caps, fmt.Sprintf("http://localhost:%d/wd/hub", port))
+		time.Sleep(time.Second)
+		if wd != nil {
+			err = nil
+			break
+		}
+	}
+
+	if err != nil {
+		panic(err)
+	}
+
+	log := utillog.GetLogger()
+
+	gob.Register(time.Time{})
+
+	// Navigate to the simple playground interface.
+	host, exists := os.LookupEnv("PORTAL_HOSTNAME")
+	if !exists {
+		host = "https://localhost:8444"
```

Comment:
This default host value should have the port declared at the top of this function.